nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/summary/ui/common/summary-table-filter/summary-table-filter.component.html

```diff
@@ -50,6 +50,21 @@
                         </mat-form-field>
                     </div>
                 }
+                @if (includeVersionedFlowStateFilter) {
+                    <div>
+                        <mat-form-field subscriptSizing="dynamic">
+                            <mat-label>Version State</mat-label>
+                            <mat-select formControlName="filterVersionedFlowState">
+                                <mat-option value="All">All States</mat-option>
+                                <mat-option value="Up to date">Up to date</mat-option>
+                                <mat-option value="Locally modified">Locally modified</mat-option>
+                                <mat-option value="Stale">Stale</mat-option>
+                                <mat-option value="Locally modified and stale">Locally modified and stale</mat-option>
+                                <mat-option value="Sync failure">Sync failure</mat-option>
+                            </mat-select>
```

Comment:
The values used should be the values found in the data returned from the API and not the  labels that the UI has assigned to them. These are enum values defined in `org.apache.nifi.registry.flow.VersionedFlowState` and probably won't change.

However, the labels used in the UI could change at any time, especially if/when NiFi supports another language.

```suggestion
                            <mat-select formControlName="filterVersionedFlowState">
                                <mat-option value="All">All States</mat-option>
                                <mat-option value="UP_TO_DATE">Up to date</mat-option>
                                <mat-option value="LOCALLY_MODIFIED">Locally modified</mat-option>
                                <mat-option value="STALE">Stale</mat-option>
                                <mat-option value="LOCALLY_MODIFIED_AND_STALE">Locally modified and stale</mat-option>
                                <mat-option value="SYNC_FAILURE">Sync failure</mat-option>
                            </mat-select>

```

This would then require a change in the filter predicate as well... see my other comment there.

Reply:
Changed to enum values